server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesCreationProcessor.java

```diff
@@ -68,147 +66,171 @@
+        if (!message.isValid()) {
+            return Either.left(new ErrorWithId(entry.getCreationId(), buildSetErrorFromValidationResult(message.validate())));
+        }
+        return Either.right(entry);
     }
+    
+    private Either<ErrorWithId, MessageWithId> handleOutboxMessages(CreationMessageEntry entry, MailboxSession session) {
+        CreationMessageId creationId = entry.getCreationId();
 
+        return getMailboxWithRole(session, Role.OUTBOX)
+            .map(outbox -> {
+                if (isRequestForSending(entry.getValue(), session)) {
+                    Function<Long, MessageId> idGenerator = uid -> generateMessageId(session, outbox, uid);
+                    return createMessageInOutboxAndSend(entry, session, outbox, idGenerator);
+                }
+                return Either.<ErrorWithId, MessageWithId>left(
+                        new ErrorWithId(creationId, 
+                                SetError.builder().description("unhandled request").type("unhandled").build()));
+                
+            }).orElse(Either.left(new ErrorWithId(creationId, UNEXPECTED_ERROR)));
+    }
+    
     @VisibleForTesting
-    protected MessageWithId<Message> createMessageInOutboxAndSend(MessageWithId.CreationMessageEntry createdEntry,
+    protected Either<ErrorWithId, MessageWithId> createMessageInOutboxAndSend(CreationMessageEntry createdEntry,
                                                            MailboxSession session,
                                                            Mailbox<Id> outbox, Function<Long, MessageId> buildMessageIdFromUid) {
+        CreationMessageId creationId = createdEntry.getCreationId();
         try {
             MessageMapper<Id> messageMapper = mailboxSessionMapperFactory.createMessageMapper(session);
             MailboxMessage<Id> newMailboxMessage = buildMailboxMessage(createdEntry, outbox);
             messageMapper.add(outbox, newMailboxMessage);
             Message jmapMessage = Message.fromMailboxMessage(newMailboxMessage, buildMessageIdFromUid);
             sendMessage(newMailboxMessage, jmapMessage, session);
-            return new MessageWithId<>(createdEntry.getCreationId(), jmapMessage);
-        } catch (MailboxException | MessagingException | IOException e) {
-            throw Throwables.propagate(e);
+            return Either.right(new MessageWithId(creationId, jmapMessage));
+            
+        } catch (MessagingException | MailboxException e) {
+            return Either.left(new ErrorWithId(creationId, UNEXPECTED_ERROR));
+            
         } catch (MailboxRoleNotFoundException e) {
             LOGGER.error("Could not find mailbox '%s' while trying to save message.", e.getRole().serialize());
-            throw Throwables.propagate(e);
+            return Either.left(new ErrorWithId(creationId, UNEXPECTED_ERROR));
         }
     }
+    
+    private boolean isMessageSetIn(Role role, CreationMessage entry, MailboxSession mailboxSession) {
```

Comment:
I don't understand messageSet. I woud prefer you to name this method isAppendToMailboxWithRole or something like this, more explicit.